test/tstDBSCAN.cpp

```diff
@@ -0,0 +1,171 @@
+#include "ArborX_DBSCANVerification.hpp"
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include <ArborX_DBSCAN.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+template <typename DeviceType, typename T>
+auto buildView(std::vector<T> const &v)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+
+  Kokkos::View<T *, DeviceType> view("Testing::v", v.size());
+  Kokkos::deep_copy(ExecutionSpace{}, view,
+                    Kokkos::View<T const *, Kokkos::HostSpace,
+                                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>(
+                        v.data(), v.size()));
+  return view;
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(dbscan_verifier, DeviceType, ARBORX_DEVICE_TYPES)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+  using ArborX::Point;
+  using ArborX::Details::verifyDBSCAN;
+
+  ExecutionSpace space;
+
+  {
+    auto points = buildView<DeviceType, Point>({{{0, 0, 0}}, {{1, 1, 1}}});
+
+    auto r = std::sqrt(3);
+
+    BOOST_TEST(verifyDBSCAN(space, points, r - 0.1, 2,
+                            buildView<DeviceType, int>({-1, -1})));
+    BOOST_TEST(!verifyDBSCAN(space, points, r - 0.1, 2,
+                             buildView<DeviceType, int>({1, 2})));
+    BOOST_TEST(!verifyDBSCAN(space, points, r - 0.1, 2,
+                             buildView<DeviceType, int>({1, 1})));
+    BOOST_TEST(
+        verifyDBSCAN(space, points, r, 2, buildView<DeviceType, int>({1, 1})));
+    BOOST_TEST(
+        !verifyDBSCAN(space, points, r, 2, buildView<DeviceType, int>({1, 2})));
+    BOOST_TEST(verifyDBSCAN(space, points, r, 3,
+                            buildView<DeviceType, int>({-1, -1})));
+    BOOST_TEST(
+        !verifyDBSCAN(space, points, r, 3, buildView<DeviceType, int>({1, 1})));
+  }
+
+  {
+    auto points = buildView<DeviceType, Point>(
+        {{{0, 0, 0}}, {{1, 1, 1}}, {{3, 3, 3}}, {{6, 6, 6}}});
+
+    auto r = std::sqrt(3);
+
+    BOOST_TEST(verifyDBSCAN(space, points, r, 2,
+                            buildView<DeviceType, int>({1, 1, -1, -1})));
+    BOOST_TEST(verifyDBSCAN(space, points, r, 3,
+                            buildView<DeviceType, int>({-1, -1, -1, -1})));
+
+    BOOST_TEST(verifyDBSCAN(space, points, 2 * r, 2,
+                            buildView<DeviceType, int>({3, 3, 3, -1})));
+    BOOST_TEST(verifyDBSCAN(space, points, 2 * r, 3,
+                            buildView<DeviceType, int>({3, 3, 3, -1})));
+    BOOST_TEST(verifyDBSCAN(space, points, 2 * r, 4,
+                            buildView<DeviceType, int>({-1, -1, -1, -1})));
+
+    BOOST_TEST(verifyDBSCAN(space, points, 3 * r, 2,
+                            buildView<DeviceType, int>({5, 5, 5, 5})));
+    BOOST_TEST(verifyDBSCAN(space, points, 3 * r, 3,
+                            buildView<DeviceType, int>({5, 5, 5, 5})));
+    BOOST_TEST(verifyDBSCAN(space, points, 3 * r, 4,
+                            buildView<DeviceType, int>({7, 7, 7, 7})));
```

Comment:
Why are the labels different for these two cases?

Reply:
The labels don't matter per say. What matters is that the points in a cluster have the same ones. Thus, I'm using some random labels to make sure that we don't rely on any particular one.

Reply:
Ah, OK. That's what I thought.